Aquarius/Src/Aquarius/Core/Input.cpp

```diff
@@ -0,0 +1,39 @@
+#include "Input.h"
+
+#include "Application.h"
+
+#include <GLFW/glfw3.h>
+
+namespace Aquarius {
+
+	KeyState getKeyState(KeyCode key)
+	{
+		auto window = Application::get()->getWindow();
+		return static_cast<KeyState>(glfwGetKey(window->get(), static_cast<int>(key)));
+	}
+
+	MouseButtonState getMouseButtonState(MouseButtonCode button)
+	{
+		auto window = Application::get()->getWindow();
+		return static_cast<MouseButtonState>(glfwGetMouseButton(window->get(), static_cast<int>(button)));
```

Comment:
This won't implicitly cast because you are using enum class and not enum right?

Reply:
Correct yeah